classes/user_table_helper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+namespace local_raisecli;
+
+/**
+ * CLI utility functions
+ *
+ * @package    local_raisecli
+ * @copyright  2022 OpenStax
+ * @license    https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Utility function to query / set a user UUID for RAISE
+ *
+ * @return string A new or existing user UUID
+ */
+class user_table_helper {
+
+    public static function get_user_table_entries($userids) {
+        global $USER, $DB;
+
+        if (count($userids) == 0) {
```

Comment:
General comment on the usage of DB APIs: Catalyst pointed this out to me in the past, but we should use `get_recordset` / `get_recordset_select` for the reasons mentioned [here](https://docs.moodle.org/dev/Data_manipulation_API#Using_recordsets).

You can find an example of using `get_recordset` in our `get_self_enrolment_methods` function.